mingw_files/nrnmingwenv.sh

```diff
@@ -1,53 +1,99 @@
-#!/bin/sh
+#!/bin/bash
+
 set -ex
 
+# Primarily (autotools and cmake) copies minimal developent files
+# to allow nrnivmodl (mknrndll) to build nrnmech.dll .
+# The -cmake second arg option does a few extra things to prepare
+# for making the setup.exe. I.e. remove some inadvertenty installed
+# file that needlessly bloat up the setup.exe, make sure that
+# dlls needed by nrniv.exe, etc, are present, copy some mswin
+# specific scripts, and make sure the neurondemo nrnmech.dll
+# is present.
+
 # if arg then assume it is destination, eg, c:/marshalnrn64/nrn
-if test $# = 1 ; then
-  N="$1/mingw"
+if test $# -ge 1 ; then
+  N="$1"
 else
-  N='c:/nrn/mingw'
-  rm -r -f "$N"
+  N='c:/nrn'
+  rm -r -f "$N/mingw"
+fi
+
+NM=$N/mingw
+
+if test "$2" = "-cmake" ; then
+  is_cmake=yes
+  srcdir="$3"
+fi
+
+if test "$is_cmake" = "yes" ; then
+  # Minimal runtime msmpi (assumes installed in c:/ms-mpi)
+  mpiinst=c:/ms-mpi
+  cp $mpiinst/bin/mpiexec.exe $N/bin
+  cp $mpiinst/bin/smpd.exe $N/bin
+  cp $mpiinst/lib/x64/msmpi.dll $N/bin
+  # These particular *.dll.a files not needed
+  rm -f tmp.tmp `find $N/bin -name \*.dll.a`
+  rm -f tmp.tmp `find $N/lib -name \*.dll.a`
+  # Inadvertently installed
+  rm -f tmp.tmp `find $N/lib -name Makefile.am`
+  rm -f tmp.tmp `find $N/lib -name \*.in`
+  # This is huge. Leave it out.
+  rm -r -f $N/lib/python/neuron/rxdtests/correct_data
+  # Significantly reduces size if compiled with -g
+  strip $N/bin/*.exe `find $N/lib -name \*.dll`
+  # til mingw vs linux straightened out replace with old mingw scripts
+  (cd $srcdir/src/mswin/bin ; cp neurondemo nrngui mknrndll $N/bin )
```

Comment:
mknrndll works for mingw and is the equivalent of nrnivmodl which does not work on mingw at present. Either mknrndll should overwrite the nrnivmodl or (a longer term project) nrnivmodl and ancillary scripts should be modified to work on mingw as well as linux/mac.

Reply:
this could be a separate ticket for later.